src/lib.rs

```diff
@@ -92,6 +95,45 @@ impl Peer {
     pub fn age(&self) -> Duration {
         self.last_seen.elapsed()
     }
+
+    /// Returns the TXT attribute data of the peer.
+    ///
+    /// This crate supports a single TXT record per peer, which contains a list
+    /// of key-value pairs of UTF-8 strings. If the value is empty, then it is a
+    /// boolean attribute, simply identified as being present, with no value.
+    ///
+    /// The formatting of the TXT record follows [RFC 6763]. Other than the RFC,
+    /// keys and values may be UTF-8 strings in addition to US-ASCII strings.
+    ///
+    /// Returns an iterator of the key-value pairs set by this peer.
+    ///
+    /// [RFC 6763]: https://datatracker.ietf.org/doc/html/rfc6763#section-6
+    pub fn txt_attributes(&self) -> impl Iterator<Item = (&str, Option<&str>)> + '_ {
+        self.txt
+            .iter()
+            .map(|(k, v)| (k.as_str(), v.as_ref().map(|v| v.as_str())))
+    }
+
+    /// Returns the value for a TXT attribute for this peer.
+    ///
+    /// See [`Self::txt_attributes`] for details.
+    ///
+    /// Returns `None` if the attribute is missing.
+    /// Returns `Some(None)` if the attribute is a boolean, i.e. has no value.
```

Comment:
Strictly speaking the attribute isn’t boolean, it has unit value or is absent. How about “if the attribute is a flag”?

Reply:
:+1: Adjusted the wording